examples/imitation_learning_openbot/run_trained_policy.py

```diff
@@ -114,69 +101,47 @@
 
         # reset the simulation
         env_reset_info = {}
-        _ = env.reset(reset_info=env_reset_info)
+        obs = env.reset(reset_info=env_reset_info)
         assert "success" in env_reset_info
 
-        # if it took too long to reset the simulation, then continue
-        if not env_reset_info["success"]:
-            spear.log("Call to env.reset(...) was not successful. Simulation took too long to return to a ready state. Skipping...")
-            prev_scene_id = episode["scene_id"]
-            continue
-
-        # send zero action to the agent and collect initial trajectory observations:
-        obs, _, _, env_step_info = env.step(action={"apply_voltage": np.array([0.0, 0.0], dtype=np.float32)})
+        # get a trajectory for this episode based on start and end point
+        episode_start_location = [[episode["start_location_x"], episode["start_location_y"], episode["start_location_z"]]]
+        episode_goal_location  = [[episode["goal_location_x"], episode["goal_location_y"], episode["goal_location_z"]]]
+        trajectory = env.get_trajectories(episode_start_location, episode_goal_location)
 
         if args.benchmark:
             start_time_seconds = time.time()
         else:
             # create dirs for storing data
-            scene_dir = os.path.realpath(os.path.join(args.eval_dir, episode["scene_id"]))
+            scene_dir   = os.path.realpath(os.path.join(args.eval_dir, episode["scene_id"]))
             episode_dir = os.path.realpath(os.path.join(scene_dir, "%04d" % episode["index"]))
-            image_dir = os.path.realpath(os.path.join(episode_dir, "images"))
-            result_dir = os.path.realpath(os.path.join(episode_dir, "results"))
-            plots_dir = os.path.realpath(os.path.join(episode_dir, "plots"))
+            image_dir   = os.path.realpath(os.path.join(episode_dir, "images"))
+            result_dir  = os.path.realpath(os.path.join(episode_dir, "results"))
+            plots_dir   = os.path.realpath(os.path.join(episode_dir, "plots"))
             os.makedirs(image_dir, exist_ok=True)
             os.makedirs(result_dir, exist_ok=True)
             os.makedirs(plots_dir, exist_ok=True)
 
             # buffer containing the state_data observations made by the agent during an episode
-            state_data = np.empty([args.num_iterations_per_episode, 6], dtype=np.float32)
+            state_data = np.empty([args.num_iterations_per_episode, 6], dtype=np.float64)
 
             # save the optimal goal trajectory in a dedicated file
-            df_trajectory = pd.DataFrame({"x_d[cm]" : [env_step_info["agent_step_info"]["trajectory_data"][:,0]],
-                                          "y_d[cm]" : [env_step_info["agent_step_info"]["trajectory_data"][:,1]],
-                                          "z_d[cm]" : [env_step_info["agent_step_info"]["trajectory_data"][:,2]]})
+            df_trajectory = pd.DataFrame({"x_d[cm]" : trajectory[0][:,0],
```

Comment:
- mimic the approach in `generate_dataset.py`, where we write the `DataFrames` together, and we do once at the end of each episode

Reply:
Updated this